ssr/src/state/canisters.rs

```diff
@@ -102,9 +102,10 @@ impl Canisters<true> {
             .expect("Authenticated canisters must have an identity")
     }
 
-    pub fn user_canister(&self) -> Principal {
-        self.user_canister
-    }
+    // Shifted to Cansiter<A>. Is this ok ?
```

Comment:
Nope,
Unauthenticated canisters don't own a user canister so its won't really return the right thing